name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/LeewayQuoteFeed.java

```diff
@@ -0,0 +1,221 @@
+package name.abuchen.portfolio.online.impl;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.text.MessageFormat;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+
+import name.abuchen.portfolio.Messages;
+import name.abuchen.portfolio.PortfolioLog;
+import name.abuchen.portfolio.model.Exchange;
+import name.abuchen.portfolio.model.LatestSecurityPrice;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityPrice;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.online.QuoteFeed;
+import name.abuchen.portfolio.online.QuoteFeedData;
+import name.abuchen.portfolio.util.WebAccess;
+
+public class LeewayQuoteFeed implements QuoteFeed
+{
+    // https://api.leeway.tech/swagger_output.json
+
+    private String apiKey;
+
+    public static final String ID = "LEEWAY"; //$NON-NLS-1$
+
+    @Override
+    public String getId()
+    {
+        return ID;
+    }
+
+    @Override
+    public String getName()
+    {
+        return "PWP Leeway UG"; //$NON-NLS-1$
+    }
+
+    public void setApiKey(String apiKey)
+    {
+        this.apiKey = apiKey;
+    }
+
+    @Override
+    public Optional<LatestSecurityPrice> getLatestQuote(Security security)
```

Comment:
Because you do essentially the same call to get the latest quote as for the historical quotes, I propose to add caching (see portfolio report). Otherwise we eat into the API quota very fast. Probably means last setting #mergeDownloadRequests to true